web/src/main/proto/spine/web/subscriptions.proto

```diff
@@ -0,0 +1,181 @@
+    //
+    // This value is a requested value. The server may set another lifetime to the subscriptions.
+    // The actual "valid through" time is returned by the server in response to this request.
+    //
+    google.protobuf.Duration lifespan = 2;
+}
+
+// The response to the `Subscribe` request.
+message Subscribing {
+
+    // The subscription results list.
+    //
+    // For each subscription topic, this is either a `Subscription`, or an error which prohibits
+    // creating such a subscription.
+    //
+    // The order of the results corresponds to the order of topics in the `Subscribe` request.
+    //
+    repeated SubscriptionOrError result = 1;
+
+    // For all the subscriptions in the response, this is the time after which the subscription
+    // will expire.
+    //
+    // An expired subscription may be cancelled by the server.
+    //
+    google.protobuf.Timestamp valid_thru = 2;
+}
+
+// Either a `spine.client.Subscription` or an error that occurred when creating the subscription.
+message SubscriptionOrError {
+
+    oneof payload {
+        WebSubscription subscription = 1;
+        base.Error error = 2;
+    }
+}
+
+// A request to prolong the life of the given subscriptions.
+message KeepUp {
+
+    // IDs of the subscriptions to keep up.
+    repeated client.SubscriptionId subscription = 1;
+
+    // How much time does the client require.
+    //
+    // The client requests that the server add this amount of time to the subscription life.
+    // The server may add this time or another duration. The updated subscription expiration date
+    // is provided in the response.
+    //
+    // If the client wishes to *reduce* the lifetime of the subscription, it can do so by providing
+    // a negative duration.
+    //
+    google.protobuf.Duration prolong_by = 2;
+}
+
+// The response to the `KeepUp` request.
+message KeepingUp {
+
+    // Outcomes of the request, one per each subscription.
+    //
+    // If an `KeepUpOutcome` contains a `new_valid_thru` time, the keep up was successful and
+    // the subscription expiration is postponed.
+    //
+    // If an `KeepUpOutcome` contains an error and the subscription exists and is active,
+    // the lifetime will remain the same.
+    //
+    repeated KeepUpOutcome outcome = 1;
+
+    // The updated expiration time for all the associated subscriptions.
+    google.protobuf.Timestamp valid_thru = 2;
+}
+
+// An outcome of a keep-up request for a single subscription.
+//
+// If the request was successful, the outcome will contain a new expiration time for
+// the subscription. The expiration time may be shorter or longer than requested. The server has
+// full control over subscription expiration.
+//
+// If the request was rejected, the `error` should explain the reason.
+//
+message KeepUpOutcome {
+
+    client.SubscriptionId id = 1;
+
+    oneof kind {
+
+        google.protobuf.Timestamp new_valid_thru = 2;
```

Comment:
Continuing on my comment somewhere above...

This should say if it's inclusive or exclusive.